src/lib/pubkey/ed25519/ed25519.h

```diff
@@ -73,16 +74,41 @@ class BOTAN_PUBLIC_API(2, 2) Ed25519_PrivateKey final : public Ed25519_PublicKey
       Ed25519_PrivateKey(const AlgorithmIdentifier& alg_id, std::span<const uint8_t> key_bits);
 
       /**
-      * Generate a private key.
+      * Generate a new random private key.
       * @param rng the RNG to use
       */
       explicit Ed25519_PrivateKey(RandomNumberGenerator& rng);
 
       /**
       * Construct a private key from the specified parameters.
+      *
+      * The behavior of this function depends on the input length.
+      *
+      * If the input is 32 bytes long then it is treated as a seed, and a new
+      * keypair is generated.
+      *
+      * If the input is 64 bytes long then it is treated as a pair of 32 byte
+      * values, first the private key and then the public key.
+      *
+      * If you are using only one of these paths, consider using instead
+      * from_seed or from_bytes.
+      *
       * @param secret_key the private key
       */
-      explicit Ed25519_PrivateKey(const secure_vector<uint8_t>& secret_key);
+      explicit Ed25519_PrivateKey(std::span<const uint8_t> secret_key);
```

Comment:
How about making this deprecated and redirect users to `from_seed` and `from_bytes`?